### PR TITLE
feat: add simulations for update credit class messages

### PR DESCRIPTION
**Pull request by aleem1314**
## Description

Closes: #556 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [x] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Review by likhita-809 — APPROVED** [review]
lgtm

File: x/ecocredit/simulation/operations.go [hunk 1]
```diff
@@ -481,6 +521,156 @@ func SimulateMsgCancel(ak ecocredit.AccountKeeper, bk ecocredit.BankKeeper,
...
+		ctx := regentypes.Context{Context: sdkCtx}
+		class, op, err := getRandomClass(ctx, r, qryClient, TypeMsgUpdateClassAdmin)
+		if class == nil {
+			return op, nil, err
+		}
+
+		addr, err := sdk.AccAddressFromBech32(class.Admin)
+		if err != nil {
+			return simtypes.NoOpMsg(ecocredit.ModuleName, TypeMsgUpdateClassAdmin, err.Error()), nil, err
+		}
```
Comment (atheeshp) [inline x/ecocredit/simulation/operations.go:539]: looks common

File: x/ecocredit/simulation/operations.go [hunk 2]
```diff
@@ -481,6 +521,156 @@ func SimulateMsgCancel(ak ecocredit.AccountKeeper, bk ecocredit.BankKeeper,
...
+		msg := &ecocredit.MsgUpdateClassAdmin{
+			Admin:    class.Admin,
+			ClassId:  class.ClassId,
+			NewAdmin: newAdmin.Address.String(),
+		}
+
+		admin, found := simtypes.FindAccount(accs, addr)
+		if !found {
+			return simtypes.NoOpMsg(ecocredit.ModuleName, TypeMsgUpdateClassAdmin, "account not found"), nil, nil
+		}
```
Comment (atheeshp) [inline x/ecocredit/simulation/operations.go:555]: same

**Review by atheeshp — COMMENTED** [review]
looks good, Is it possible to move the common code to a util?

File: x/ecocredit/simulation/operations.go [hunk 3]
```diff
@@ -481,6 +509,155 @@ func SimulateMsgCancel(ak ecocredit.AccountKeeper, bk ecocredit.BankKeeper,
...
+func SimulateMsgUpdateClassIssuers(ak ecocredit.AccountKeeper, bk ecocredit.BankKeeper,
+	qryClient ecocredit.QueryClient) simtypes.Operation {
+	return func(
+		r *rand.Rand, app *baseapp.BaseApp, sdkCtx sdk.Context, accs []simtypes.Account, chainID string,
+	) (simtypes.OperationMsg, []simtypes.FutureOperation, error) {
+		ctx := regentypes.Context{Context: sdkCtx}
+		class, op, err := getRandomClass(ctx, r, qryClient, TypeMsgUpdateClassIssuers)
+		if class == nil {
+			return op, nil, err
+		}
```
Comment (atheeshp) [inline x/ecocredit/simulation/operations.go:611]: Can't we move this part as well into the util?

**Review by atheeshp — APPROVED** [review]
looks good

File: x/ecocredit/simulation/operations.go (hunk 3 above)
Comment (aleem1314) [reply x/ecocredit/simulation/operations.go:611]: We need batch-info in other operations([SimulateMsgRetire](https://github.com/regen-network/regen-ledger/blob/master/x/ecocredit/simulation/operations.go#L332-L340)) before `getAccountAndSpendableCoins`.

**Pull request merged**